translated code of Python.h fails to build

### Zig Version

0.9.1

### Steps to Reproduce

On Ubuntu 20.04 with python 3.8 installed
```
$ zig translate-c /usr/include/python3.8/Python.h -I/usr/include/python3.8 -I /usr/include -I /usr/include/x86_64-linux-gnu/ > python3.8.zig
$ zig build-lib python3.8.zig
```

this is also tested on `0.10.0-dev.1091+7fc8dd664`. 

`@cImport` works with no issues. (check here https://github.com/ali5h/zig-py-sample)

### Expected Behavior

translated code should compile

### Actual Behavior

```
$ zig build-lib python3.8.zig
python3.8.zig:9061:124: error: redefinition of label 'blk'
        break :blk if (@import("std").zig.c_translation.cast([*c]PyASCIIObject, op).*.wstr) PyUnicode_WSTR_LENGTH(op) else blk: {
                                                                                                                           ^
python3.8.zig:9059:12: note: previous definition here
    return blk: {
           ^
python3.8.zig:9516:30: error: use of undeclared identifier 'sched_priority'
pub const __sched_priority = sched_priority;
```

--- a/src/translate_c.zig
+++ b/src/translate_c.zig
@@ -5563,7 +5563,7 @@ fn parseCExpr(c: *Context, m: *MacroCtx, scope: *Scope) ParseError!Node {
         const ignore = try Tag.discard.create(c.arena, .{ .should_skip = false, .value = last });
         try block_scope.statements.append(ignore);
 
-        last = try parseCCondExpr(c, m, scope);
+        last = try parseCCondExpr(c, m, &block_scope.base);
         if (m.next().? != .Comma) {
             m.i -= 1;
             break;

--- a/test/behavior/translate_c_macros.h
+++ b/test/behavior/translate_c_macros.h
@@ -37,3 +37,5 @@ union U {
 #define IGNORE_ME_10(x) (volatile const void)(x)
 
 #define UNION_CAST(X) (union U)(X)
+
+#define NESTED_COMMA_OPERATOR (1, (2, 3))

--- a/test/behavior/translate_c_macros.zig
+++ b/test/behavior/translate_c_macros.zig
@@ -56,3 +56,7 @@ test "casting to union with a macro" {
     casted = h.UNION_CAST(d);
     try expectEqual(d, casted.d);
 }
+
+test "nested comma operator" {
+    try expectEqual(@as(c_int, 3), h.NESTED_COMMA_OPERATOR);
+}

--- a/test/behavior/translate_c_macros.zig
+++ b/test/behavior/translate_c_macros.zig
@@ -58,5 +58,7 @@ test "casting to union with a macro" {
 }
 
 test "nested comma operator" {
+    if (builtin.zig_backend == .stage2_c) return error.SkipZigTest; // TODO
+
     try expectEqual(@as(c_int, 3), h.NESTED_COMMA_OPERATOR);
 }

--- a/test/run_translated_c.zig
+++ b/test/run_translated_c.zig
@@ -1851,4 +1851,14 @@ pub fn addCases(cases: *tests.RunTranslatedCContext) void {
         \\    return 0;
         \\}
     , "");
+
+    cases.add("Nested comma operator in macro. Issue #11040",
+        \\#include <stdlib.h>
+        \\#define FOO (1, (2,  3))
+        \\int main(void) {
+        \\    int x = FOO;
+        \\    if (x != 3) abort();
+        \\    return 0;
+        \\}
+    , "");
 }

